packages/@aws-cdk/aws-ecs-patterns/test/ec2/test.queue-processing-ecs-service.ts

```diff
@@ -80,6 +81,34 @@ export = {
     test.done();
   },
 
+  'test ECS queue worker service construct - with remove default desiredCount feature flag'(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+    stack.node.setContext(cxapi.ECS_REMOVE_DEFAULT_DESIRED_COUNT, true);
+
+    const vpc = new ec2.Vpc(stack, 'VPC');
+    const cluster = new ecs.Cluster(stack, 'Cluster', { vpc });
+    cluster.addCapacity('DefaultAutoScalingGroup', { instanceType: new ec2.InstanceType('t2.micro') });
+
+    // WHEN
+    new ecsPatterns.QueueProcessingEc2Service(stack, 'Service', {
+      cluster,
+      memoryLimitMiB: 512,
+      image: ecs.ContainerImage.fromRegistry('test'),
+    });
+
+    // THEN - QueueWorker is of EC2 launch type, an SQS queue is created and all default properties are set.
+    expect(stack).to(haveResource('AWS::ECS::Service', {
+      LaunchType: 'EC2',
+    }));
+
+    expect(stack).notTo(haveResource('AWS::ECS::Service', {
+      DesiredCount: 1,
+    }));
+
+    test.done();
+  },
```

Comment:
```suggestion
    expect(stack).to(haveResource('AWS::ECS::Service', {
      LaunchType: 'EC2',
      DesiredCount: ABSENT,
    }));
    
    test.done();
  },
```

Reply:
The ec2 module actually uses both to and notTo, rather than ABSENT. I was trying to stick to what's already there.

Reply:
why does the ec2 module matter? was it asserting on a property being absent or a resource?

Reply:
I wasn't trying to call out that the ec2 module mattered, but rather that's how the cdk currently does it in other modules.

It's asserting on properties both existing and being absent: https://github.com/aws/aws-cdk/blame/master/packages/%40aws-cdk/aws-applicationautoscaling/test/test.scalable-target.ts#L113-L119

Reply:
I don't think it matters much, @SoManyHs suggestion is my preference.